### DISCUSSION TITLE
How to set globak keyboard shortcut remap? Like "Caps" -> "Meta + Space"

### DISCUSSION THREAD

**Discussion by linonetwo** [pop-os/cosmic-epoch#1382] [Q&A]
On MacOS and Windows11, I can remap Caps Lock (rarely used) to any other keyboard shortcuts, like map it to sequental two keys "Meta + Space".

And I will use "Meta + Space" to switch IME.

Can this be done in Cosmic rust-based desktop environment?

**Comment by dazu89** [comment]
I am also interested to know how key remap works in cosmic – the caps lock key is so conveniently reached with my stubby fingers!

**Comment by JosXa** [comment]
I have this in my NixOS config, maybe it'll hint you at what you need to do on your end (excerpt done by opencode):

## Primary Method: xremap Service
The main way you remapped Caps Lock to Escape was through **xremap** (`home-manager/desktop.nix:92-97`):
```nix
home.file.".config/xremap/config.yml".text = ''
  modmap:
    - name: Global
      remap:
        CapsLock: Esc
'';
```
This is coupled with a systemd user service (`home-manager/desktop.nix:169-180`) that auto-starts xremap.
## Desktop Environment XKB Configuration
You also configured keyboard remapping directly in desktop environments:
- **COSMIC**: XKB options in `.config/cosmic/com.system76.CosmicComp/v1/xkb_config` with `caps:escape`
- **GNOME**: dconf settings with `xkb-options = ["caps:escape"]`

**Comment by dazu89** [comment]
Regarding Cosmic and xremap, while global shortcuts ought to work (not tested) support for application specific shortcuts have not been implemented https://github.com/xremap/xremap/discussions/635.
